pkg/proxy/dns.go

```diff
@@ -9,11 +9,17 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/cilium/cilium/pkg/fqdn/defaultdns"
+	"github.com/cilium/cilium/pkg/identity"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 	"github.com/cilium/cilium/pkg/policy"
 	"github.com/cilium/cilium/pkg/revert"
 )
 
+var (
+	// GlobalStandaloneDNSProxy is the global, shared, standalone DNS Proxy singleton.
+	GlobalStandaloneDNSProxy sdpPolicyUpdater
```

Comment:
Global singletons are not allowed as this destroys ability to do parallel testing. This must be passed as argument to where it is needed.

Reply:
It also seems this is really easy to pass this in here:
- Add the "PolicyUpdater" as exported interface to the `pkg/fqdn/service`
- Refactor pkg/fqdn/service to be a properly implemented as a module
- Have a provider function in `pkg/fqdn/service` provide that type to the application
- Depend on `service.PolicyUpdater` in `proxyParams`

Reply:
I updated it the way you recommended. Let me know if it is correct.